packages/tests/src/_jest/helpers/mocks.ts

```diff
@@ -64,6 +65,46 @@ const logFn: Logger = {
 };
 export const mockLogger: Logger = logFn;
 
+export const getEsbuildMock = (options: Partial<PluginBuild> = {}): PluginBuild => {
+    return {
+        resolve: async (filepath) => {
+            return {
+                errors: [],
+                warnings: [],
+                external: false,
+                sideEffects: false,
+                namespace: '',
+                suffix: '',
+                pluginData: {},
+                path: getAbsolutePath(process.cwd(), filepath),
+            };
+        },
+        onStart: jest.fn(),
+        onEnd: jest.fn(),
+        onResolve: jest.fn(),
+        onLoad: jest.fn(),
+        onDispose: jest.fn(),
+        ...options,
```

Comment:
You may want to move `...options` to the 1st element to avoid being able to override the other fields. And if you want that to be its behavior, you should add a comment saying "All the methods before can be overridden in `options`" or something like that

Reply:
Changed `options` for `overrides`.